catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java

```diff
@@ -193,13 +203,27 @@ public Metacard getMetacard(String id)
         }
 
         Filter queryFilter = filterBuilder.anyOf(filters);
-        QueryResponse response = catalogFramework.query(new QueryRequestImpl(new QueryImpl(
-                queryFilter,
-                1,
-                -1,
-                SortBy.NATURAL_ORDER,
-                false,
-                TimeUnit.SECONDS.toMillis(10)), false));
+        QueryRequest queryRequest;
+
+        if (CollectionUtils.isEmpty(writableSources)) {
+            queryRequest = new QueryRequestImpl(new QueryImpl(
+                    queryFilter,
+                    1,
+                    querySize,
```

Comment:
❓ For this query and the one in the `else` block, should we consider waiting for https://github.com/codice/ddf/pull/2080 to get merged and use the new result iterator/pagination utility? That way we wouldn't need to ever worry about `querySize` getting too big or capped.